src/OpeningHoursForDay.php

```diff
@@ -93,7 +93,11 @@ public function nextClose(Time $time)
 
     protected function findNextOpenInWorkingHours(Time $time, TimeRange $timeRange)
     {
-        if ($timeRange->containsTime($time) && next($timeRange) !== $timeRange) {
+        if ($timeRange->containsTime($time) && $next = next($timeRange) !== $timeRange) {
```

Comment:
Due to operators precedence, `$next` will always be a boolean (in this case, `false`), so `! $next instanceof Time` will always be `true`. We are just always skipping a value here as far as I understand. I will try to create a unit test to confirm it.

Reply:
Yes, you are right.. the strangest thing is that if you just return false from this method all tests will still pass.

```php
protected function findNextOpenInWorkingHours(Time $time, TimeRange $timeRange)
{
    return false;
}
```